src/core/include/openvino/core/op_extension.hpp

```diff
@@ -33,12 +33,36 @@ class OPENVINO_API BaseOpExtension : public Extension {
      */
     virtual ov::OutputVector create(const ov::OutputVector& inputs, ov::AttributeVisitor& visitor) const = 0;
 
+    /**
+     * @brief Returns extensions that should be registered together with this extension class object.
+     *
+     * Attached extensions may include frontend extensions that OpenVINO op to framework ops or necessary
+     * transformations that should be applied to the network which consist of target op.
+     *
+     * @return
+     */
+    virtual std::vector<ov::Extension::Ptr> get_attached_extensions() const = 0;
```

Comment:
should not `attached_extensions` be registered before current extension is added to the FE / Core? Instead of hierarchy of extensions

Reply:
What do you mean? 
To support this syntax, when we register a new frontend::OpExtension via OPENVINO_FRAMEWORK_MAP(...) macro we need to collect all in core::OpExtension method, because the user type, in this case CustomOp, is known only on this level. 
![image](https://user-images.githubusercontent.com/21181928/152347206-0ab11c5d-e5ba-4bd7-81bf-8296320a8f74.png)

full pipeline looks like that:
UserClass - wrapped to -> core::OpExtension(UserClass) - casted to -> core::BaseOpExtension  - calls -> virtual method attached_extension() - collects -> frontend::OpExtension registered via OPENVINO_FRAMEWORK_MAP 
after that frontend::OpExtension are passed to each FrontEnd